### PR TITLE
[util] Add templates to autogen ISR testutils

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds two templates to autogenerate ISR testutil implementations (in C) and a meson.build file to build the ISR testutil library. Additionally, this commit updates the `util/autogen_testutils.py` script to function with the new templates.

This partially addresses #9038.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

File: util/autogen_testutils.py [hunk 1]
```diff
@@ -36,13 +39,33 @@ def main():
             "ERROR: clang-format must be installed to format "
             " autogenerated code to pass OpenTitan CI checks.")
 
+    # Parse command line args.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--topcfg", "-t", help="path of the top hjson file.")
+    args = parser.parse_args()
+
+    # Parse toplevel Hjson to get IPs that are templated / generated with IPgen.
+    topcfg_path = REPO_TOP / "hw/top_earlgrey/data/top_earlgrey.hjson"
```
Comment (mcy) [inline util/autogen_testutils.py:48]: wouldn't it make more sense to use `default=` in the parser?

File: util/autogen_testutils/isr_testutils.c.tpl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    #include "sw/device/lib/dif/dif_${ip.name_snake}.h"
+  % endif
+% endfor
+#include "sw/device/lib/testing/check.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h" // Generated.
+
+% for ip in ips_with_difs:
+  % if ip.irqs:
+    void isr_testutils_${ip.name_snake}_isr(dif_rv_plic_t *rv_plic,
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply util/autogen_testutils/isr_testutils.c.tpl:21]: Good idea. Fixed.

File: util/autogen_testutils.py [hunk 3]
```diff
@@ -36,13 +39,33 @@ def main():
...
+    if args.topcfg:
+        topcfg_path = args.topcfg
+    try:
+        with open(topcfg_path, 'r') as ftop:
+            topcfg = hjson.load(ftop, use_decimal=True)
+    except FileNotFoundError:
+        print(f"hjson {topcfg_path} could not be found")
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply util/autogen_testutils.py:55]: Fixed.

File: util/autogen_testutils.py [hunk 4]
```diff
@@ -51,14 +74,18 @@ def main():
...
+            Ip(ip_name_snake, "AUTOGEN", templated_modules, ipgen_modules))
     ips_with_difs.sort(key=lambda ip: ip.name_snake)
 
     # Create output directories if needed.
     autogen_testutils_dir.mkdir(exist_ok=True)
 
     # Auto-generate testutils files.
-    for suffix in [".h", ".c"]:
+    for suffix in [".h", ".c", ".build"]:
+        comment_syntax = "//"
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply util/autogen_testutils.py:86]: Fixed

File: util/autogen_testutils.py [hunk 5]
```diff
@@ -36,13 +39,33 @@ def main():
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--topcfg", "-t", help="path of the top hjson file.")
+    args = parser.parse_args()
+
+    # Parse toplevel Hjson to get IPs that are templated / generated with IPgen.
+    topcfg_path = REPO_TOP / "hw/top_earlgrey/data/top_earlgrey.hjson"
+    if args.topcfg:
+        topcfg_path = args.topcfg
+    try:
+        with open(topcfg_path, 'r') as ftop:
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply util/autogen_testutils.py:52]: Fixed.

File: util/autogen_testutils.py (hunk 1 above)
Comment (timothytrippel) [reply util/autogen_testutils.py:48]: Fixed.

**Review by mcy — APPROVED** [review]

**Pull request merged**